### PR TITLE
Add ability to check for DeferrableProvider interface

### LINKED ISSUE #3: Support 5.8 Deferred Service Provider Syntax
In Laravel 5.8, a service provider won't be marked by setting a `$deferred` property, but will instead be marked by implementing the `DeferrableProvider` interface.

https://laravel.com/docs/5.8/providers#deferred-providers

In order to support 5.8, we should update the deferred provider sniff to also check any service providers that implement this new interface.

### PR DESCRIPTION
**Pull request by nathanheffley**
This PR adds detection of deferred service providers in the new 5.8 syntax, fixes #3

### CODE REVIEW

**Comment by nathanheffley** [comment]
Before this gets merged in I think I will add some configurable properties that allow the user to:
1. Set the name of the interface it searches for from the default `DeferrableProvider` to a custom string, mostly because having hard coded strings like it is now is kind of lame, and maybe somebody out there implements a custom interface on their service providers 🤷‍♂ 
2. Provide a string to search for as `extends` in addition to the `implements` and `$defer`, so that if somebody has a custom service provider that they extend which implements `DeferrableProvider` they have a way to use this package.

**Comment by DynamoGeek** [comment]
I've giving this a nice review now.

If possible, I'd prefer one or two more PRs to add those features. This is a sweet feature that, barring any issue, we could merge in pretty much right now. 😄

File: DealerInspireLaravelCodingStandard/Sniffs/Providers/DeferredProvidersSniff.php [hunk 1]
```diff
@@ -116,21 +121,33 @@ public function process(File $phpcsFile, $stackPtr): void
...
             $this->checkingForDeferredValue = true;
+            return null;
         }
 
-        if ($this->checkingForDeferredValue === false) {
-            return null;
+        if ($this->checkingForDeferrableProvider) {
+            // The first string after 'implements' will be the value
+            if ($tokens[$index]['code'] === T_STRING) {
+                $this->checkingForDeferrableProvider = false;
```
Comment (DynamoGeek) [inline DealerInspireLaravelCodingStandard/Sniffs/Providers/DeferredProvidersSniff.php]: If we search until we find a T_OPEN_CURLY_BRACKET, we could also allow this to detect when DeferrableProvider is the 2nd+ interface being implemented.

Something akin to 
```
if ($this->checkingForDeferrableProvider) {
    if ($tokens[$index]['code'] === T_OPEN_CURLY_BRACKET) {
        // We've searched all the implemented interfaces
        $this->checkingForDeferrableProvider = false;
        return null;
    }
    if ($tokens[$index]['code'] === T_STRING && $tokens[$index]['content'] === 'DeferrableProvider') {
        $this->checkingForDeferrableProvider = false;
        return true;
    }
}
```

Comment (nathanheffley) [reply DealerInspireLaravelCodingStandard/Sniffs/Providers/DeferredProvidersSniff.php]: That's really smart, I'll implement that here soon!

**Pull request merged**